### PR TITLE
add some unit tests for pkg/api

### PR DESCRIPTION
**Pull request by qiuming520**
Signed-off-by: qiuming520 <qiuming_yewu@cmss.chinamobile.com>

### What this PR does / why we need it:
add some unit tests for pkg/api
### Which issue(s) this PR fixes:

Fixes #

### Special notes for reviewers:
```

```

### Does this PR introduced a user-facing change?

```
None
```

### Additional documentation, usage docs, etc.:

```

```

### CODE REVIEW

File: pkg/api/alerting/v2alpha1/types_test.go [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+		Items: []*BulkItemResponse{
+			&succBulkItemResponse,
+			&errBulkItemResponse,
+		},
+	}
+)
+
+func TestParseAlertingRuleQueryParams(t *testing.T) {
+	queryParam := "/alerting.kubesphere.io/v2alpha1/rules?name=test-cpu&state=firing&health=ok&sort_field=lastEvaluation&sort_type=desc&label_filters=name~test"
+	req, err := http.NewRequest("GET", fmt.Sprintf("http://localhost?%s", queryParam), nil)
```
Comment (xyz-li) [inline pkg/api/alerting/v2alpha1/types_test.go:314]: Is this a redundant `?`?

Comment (qiuming520) [reply pkg/api/alerting/v2alpha1/types_test.go:314]: Obtain the http.Request object required by the restful.NewRequest function through the http.NewRequest function (various RULs) . The restful.NewRequest function returns restful.Request , which is the parameter required by the ParseQueryParameter function.  it's necessary.

File: pkg/api/alerting/v2alpha1/types_test.go [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+	bulkResponse = BulkResponse{
+		Items: []*BulkItemResponse{
+			&succBulkItemResponse,
+			&errBulkItemResponse,
+		},
+	}
+)
+
+func TestParseAlertingRuleQueryParams(t *testing.T) {
+	queryParam := "/alerting.kubesphere.io/v2alpha1/rules?name=test-cpu&state=firing&health=ok&sort_field=lastEvaluation&sort_type=desc&label_filters=name~test"
```
Comment (wansir) [inline pkg/api/alerting/v2alpha1/types_test.go:313]: `/alerting.kubesphere.io/v2alpha1/rules` is the request path (relative paths may omit leading slash).

> An HTTP URL takes the form:
> http://<host>:<port>/<path>?<searchpart>
> where <host> and <port> are as described in Section 3.1. If :<port>
is omitted, the port defaults to 80.  No user name or password is
allowed.  <path> is an HTTP selector, and <searchpart> is a query
string. The <path> is optional, as is the <searchpart> and its
preceding "?". If neither <path> nor <searchpart> is present, the "/"
may also be omitted.

https://www.rfc-editor.org/rfc/rfc1738#section-3.3

Comment (qiuming520) [reply pkg/api/alerting/v2alpha1/types_test.go]: Right! Thanks for your review, I have fixed it 。

File: pkg/api/alerting/v2alpha1/types_test.go (hunk 1 above)
Comment (wansir) [inline pkg/api/alerting/v2alpha1/types_test.go:314]:
```suggestion
	queryParam := "name=test-cpu&state=firing&health=ok&sort_field=lastEvaluation&sort_type=desc&label_filters=name~test"
	req, err := http.NewRequest("GET", fmt.Sprintf("http://localhost/alerting.kubesphere.io/v2alpha1/rules?%s", queryParam), nil)
```

Comment (qiuming520) [reply pkg/api/alerting/v2alpha1/types_test.go]: Thanks for suggestion ！ fixed and committed， help review again plz @wansir

**Pull request closed without merging**